src/trans-word-level/sequence.cpp

```diff
@@ -279,6 +279,74 @@ sequence_matchest instantiate_sequence(
 
     return result;
   }
+  else if(expr.id() == ID_sva_sequence_goto_repetition)
+  {
+    // The 'op' is a Boolean condition, not a sequence.
+    auto &op = to_sva_sequence_goto_repetition_expr(expr).op();
+    auto repetitions = numeric_cast_v<mp_integer>(
+      to_sva_sequence_goto_repetition_expr(expr).repetitions());
+    PRECONDITION(repetitions >= 1);
+
+    sequence_matchest result;
+
+    // We add up the number of matches of 'op' starting from
+    // timeframe u, until the end of our unwinding.
+    auto bits = address_bits(no_timeframes);
+    auto zero = from_integer(0, unsignedbv_typet{bits});
+    auto one = from_integer(1, unsignedbv_typet{bits});
+    exprt matches = zero;
+
+    for(mp_integer u = t; u < no_timeframes; ++u)
+    {
+      // match of op in timeframe u?
+      auto rec_op = instantiate(op, u, no_timeframes);
+
+      // add up
+      matches = plus_exprt{matches, if_exprt{rec_op, one, zero}};
+
+      // We have a match for op[->n] if there is a match in timeframe
+      // u and matches is n.
+      result.emplace_back(
+        u,
+        and_exprt{
+          rec_op,
+          equal_exprt{matches, from_integer(repetitions, matches.type())}});
```

Comment:
How about creating `from_integer(repetitions, matches.type()` (which is `from_integer(repetitions, unsignedbv_typet{bits})`) just once?

Reply:
done